diff --git a/engine/Makefile b/engine/Makefile
index da9bf1b5..60d4fd9d 100644
--- a/engine/Makefile
+++ b/engine/Makefile
@@ -202,6 +202,25 @@ BUILD_DEBUG     = 1
 endif
 endif
 
+ifdef BUILD_MMIYOO
+TARGET          = $(VERSION_NAME).elf
+TARGET_FINAL    = $(VERSION_NAME)
+TARGET_PLATFORM = MMIYOO
+BUILD_LINUX     = 1
+BUILD_SDL       = 1
+BUILD_GFX       = 1
+BUILD_PTHREAD   = 1
+BUILD_SDL_IO    = 1
+BUILD_TREMOR    = 1
+BUILDING        = 1
+INCLUDES        = ./source/webmlib \
+                  ./android/app/jni/openbor/include
+OBJTYPE         = elf
+ifeq ($(BUILD_MMIYOO), 0)
+BUILD_DEBUG     = 1
+endif
+endif
+
 ifdef BUILD_WII
 TARGET          = $(VERSION_NAME).elf
 TARGET_MAP      = $(TARGET).map
@@ -249,6 +268,9 @@ endif
 ifdef BUILD_PANDORA
 STRIP           = $(PNDDEV)/bin/arm-none-linux-gnueabi-strip $(TARGET) -o $(TARGET_FINAL)
 endif
+ifdef BUILD_MMIYOO
+STRIP           ?= arm-linux-gnueabihf-strip $(TARGET) -o $(TARGET_FINAL)
+endif
 ifdef BUILD_WII
 ELF2DOL         = elf2dol $< $@
 endif
@@ -534,6 +556,10 @@ CFLAGS         += -DLINUX
 endif
 
 
+ifdef BUILD_MMIYOO
+CFLAGS         += -DLINUX
+endif
+
 ifdef BUILD_WII
 CFLAGS        += -DCACHE_BACKGROUNDS -DREVERSE_COLOR -D__ppc__ $(MACHDEP) -Wl,-Map,$(TARGET_MAP),-wrap,wiiuse_register
 # This allows to compile with DevkitPPC r29 and above.
@@ -617,5 +643,9 @@ ifeq ($(findstring DGE, $(SDKPATH)), DGE)
 LIBS           += -lSDL
 else
+ifdef BUILD_MMIYOO
+LIBS           += -lSDL2
+else
 COMMA		=,
 RPATH_LIBS	= $(addprefix -Wl$(COMMA)-rpath$(COMMA),"$(LIBRARIES)")
 LIBS	       += $(RPATH_LIBS) -lSDL2
+endif
@@ -790,6 +820,7 @@ all:
 	@echo make BUILD_WIN=1
 	@echo make BUILD_PANDORA=1
 	@echo make BUILD_LINUX=1
+	@echo make BUILD_MMIYOO=1
 	@echo
 	@echo Cleanup Intermediate Files:
 	@echo
diff --git a/engine/sdl/control.c b/engine/sdl/control.c
index 110e264d..796b1840 100644
--- a/engine/sdl/control.c
+++ b/engine/sdl/control.c
@@ -220,7 +220,7 @@ void control_init()
     // initialize all devices to DEVICE_TYPE_NONE
     memset(devices, 0, sizeof(devices));
 
-    int numJoysticks = SDL_NumJoysticks();
+    int numJoysticks = 0;//SDL_NumJoysticks();
 
     if (numJoysticks >= MAX_DEVICES)
     {
@@ -299,15 +299,15 @@ static void set_default_keyboard_mappings(InputDevice *device)
     device->mappings[SDID_MOVEDOWN]   = SDL_SCANCODE_DOWN;
     device->mappings[SDID_MOVELEFT]   = SDL_SCANCODE_LEFT;
     device->mappings[SDID_MOVERIGHT]  = SDL_SCANCODE_RIGHT;
-    device->mappings[SDID_ATTACK]     = SDL_SCANCODE_A;
-    device->mappings[SDID_ATTACK2]    = SDL_SCANCODE_S;
-    device->mappings[SDID_ATTACK3]    = SDL_SCANCODE_Z;
-    device->mappings[SDID_ATTACK4]    = SDL_SCANCODE_X;
-    device->mappings[SDID_JUMP]       = SDL_SCANCODE_D;
-    device->mappings[SDID_SPECIAL]    = SDL_SCANCODE_F;
+    device->mappings[SDID_ATTACK]     = SDL_SCANCODE_SPACE;
+    device->mappings[SDID_ATTACK2]    = SDL_SCANCODE_LSHIFT;
+    device->mappings[SDID_ATTACK3]    = SDL_SCANCODE_E;
+    device->mappings[SDID_ATTACK4]    = SDL_SCANCODE_T;
+    device->mappings[SDID_JUMP]       = SDL_SCANCODE_LCTRL;
+    device->mappings[SDID_SPECIAL]    = SDL_SCANCODE_LALT;
     device->mappings[SDID_START]      = SDL_SCANCODE_RETURN;
     device->mappings[SDID_SCREENSHOT] = SDL_SCANCODE_F12;
-    device->mappings[SDID_ESC]        = SDL_SCANCODE_ESCAPE;
+    device->mappings[SDID_ESC]        = SDL_SCANCODE_RCTRL;
 }
 
 static void set_default_controller_mappings(InputDevice *device)
@@ -468,6 +468,11 @@ static void handle_events()
 
             case SDL_KEYDOWN:
             {
+                if(ev.key.keysym.sym == SDLK_ESCAPE){
+                    borShutdown(0, DEFAULT_SHUTDOWN_MESSAGE);
+                    break;
+                }
+
                 if (remapDevice && remapDevice->deviceType == DEVICE_TYPE_KEYBOARD)
                 {
                     remapKeycode = ev.key.keysym.scancode;
diff --git a/engine/sdl/menu.c b/engine/sdl/menu.c
index de60c0c2..00af06f5 100644
--- a/engine/sdl/menu.c
+++ b/engine/sdl/menu.c
@@ -701,17 +701,19 @@ static void drawLogs()
 
 static void drawLogo()
 {
-    int i;
-    int delay = 500;
+    //int i;
+    //int delay = 500;
 
     if(savedata.logo) return;
 
 	initMenu(0);
 
-	for(i = 0; i < delay; i++)
+	//for(i = 0; i < delay; i++)
     {
         blit_video_menu(bgscreen);
-        SDL_Delay(1);
+        SDL_Delay(1000);
+        blit_video_menu(bgscreen);
+        SDL_Delay(1000);
     }
 
 	termMenu();
diff --git a/engine/sdl/video.c b/engine/sdl/video.c
index 0f13f308..4dc1757a 100644
--- a/engine/sdl/video.c
+++ b/engine/sdl/video.c
@@ -135,7 +135,7 @@ int SetVideoMode(int w, int h, int bpp, bool gl)
 
 	if(!gl)
 	{
-		renderer = SDL_CreateRenderer(window, -1, savedata.vsync ? SDL_RENDERER_PRESENTVSYNC : 0);
+		renderer = SDL_CreateRenderer(window, -1, SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC);
 		if(!renderer)
 		{
 			printf("Error: failed to create renderer: %s\n", SDL_GetError());
